### PR TITLE
Remove PROTOTYPE from SearchSourceBuilder

### PR DESCRIPTION
**Pull request by nik9000**
Relates to #17085

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java [hunk 1]
```diff
@@ -102,16 +102,12 @@
...
 
-    private static final SearchSourceBuilder PROTOTYPE = new SearchSourceBuilder();
-
-    public static SearchSourceBuilder readSearchSourceFrom(StreamInput in) throws IOException {
-        return PROTOTYPE.readFrom(in);
-    }
-
     public static SearchSourceBuilder parseSearchSource(XContentParser parser, QueryParseContext context,
                                                         AggregatorParsers aggParsers, Suggesters suggesters)
             throws IOException {
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java]: nit: I wonder if we should rename this to "fromXContent" like we do in most of the queries, suggesters etc... No strong opinion here though.

**Comment by cbuescher** [comment]
@nik9000 left one minor comment about naming, but other than that LGTM.

**Comment by nik9000** [comment]
Thanks @cbuescher !

**Comment by nik9000** [comment]
Thanks @cbuescher !

**Pull request merged**